core/src/autogluon/core/utils/utils.py

```diff
@@ -21,86 +21,95 @@
 from ..constants import BINARY, REGRESSION, MULTICLASS, SOFTCLASS, QUANTILE
 from ..metrics import accuracy, root_mean_squared_error, pinball_loss, Scorer
 
-logger = logging.getLogger(__name__)
-
 
+logger = logging.getLogger(__name__)
 
-# TODO: RespirceManager to replace these two hacks
-class _get_cpu_count:
-    def __init__(self):
-        self.mock = None
 
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
+class ResourceManager():
+    """Manager that fetches system related info"""
+    
+    @staticmethod
+    def get_cpu_count():
         return multiprocessing.cpu_count()
     
-
-class _get_gpu_count_all:
-    """
-    Attempts to get number of GPUs available for use via multiple means.
-    """
-    # FIXME: update to use only torch for TIMM or find a better GPU detection strategy
-    # FIXME: get_gpu_count by itself doesn't always work for Windows
-    def __init__(self):
-        self.mock = None
-
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
-        num_gpus = _get_gpu_count_cuda()
+    @staticmethod
+    def get_cpu_count_psutil(logical=True):
+        return psutil.cpu_count(logical=logical)
+    
+    @staticmethod
+    def get_gpu_count_all():
```

Comment:
curious why we suffix the func name with `all`? it would be better to make it consistent with cpu func

Reply:
I think this function means get gpu count from all possible sources.

Reply:
For GPU there is not a single source of truth as to how many GPUs there are, as it depends on the package if it can actually detect and leverage the GPUs. We might consider removing the `_all` part of the name, but maybe after v0.6 release.